test/framework/fixtures.go

```diff
@@ -38,67 +42,100 @@ func NewTestFixture(name, k8sContext string) *TestFixture {
+	kustomization := &fixtureKustomization{
+		Namespace: namespace,
+		Fixture:   fixture.Name,
+		Contexts:  contexts,
+		Zones:     zones,
+		Storage:   storage,
+	}
+	if k8c, err := getFixtureK8ssandraCluster(fixture); err != nil {
+		return nil, err
+	} else if k8c != nil {
+		kustomization.K8ssandraCluster = &k8c.Spec
+	}
+	if dc, err := getFixtureCassandraDatacenter(fixture); err != nil {
+		return nil, err
+	} else if dc != nil {
+		kustomization.CassandraDatacenter = &dc.Spec
+	}
+	if sg, err := getFixtureStargate(fixture); err != nil {
+		return nil, err
+	} else if sg != nil {
+		kustomization.Stargate = &sg.Spec
+	}
+	return kustomization, nil
 }
 
-// countK8ssandraDatacenters counts the number of dc definitions in the K8ssandraCluster resource of
-// this fixture. For now, we only read one single K8ssandraCluster declared in k8ssandra.yaml. If
-// some fixtures in the future decide to create more than one K8ssandraCluster, we'll have to
-// revisit this and create per-K8ssandraCluster kustomizations.
-func (f *TestFixture) countK8ssandraDatacenters() (int, error) {
-	k8ssandraYamlFile := filepath.Join(f.definitionsDir, "k8ssandra.yaml")
-	if _, err := os.Stat(k8ssandraYamlFile); err == nil {
-		result, err := yq.Eval(
-			". | select(.kind == \"K8ssandraCluster\") | .spec.cassandra.datacenters.[] as $item ireduce (0; . +1)",
-			yq.Options{All: true},
-			k8ssandraYamlFile,
-		)
-		if err != nil {
-			return -1, err
-		}
-		return strconv.Atoi(result)
-	} else if errors.Is(err, fs.ErrNotExist) {
-		return 0, nil
-	} else {
-		return -1, err
+func getFixtureK8ssandraCluster(fixture *TestFixture) (*coreapi.K8ssandraCluster, error) {
+	obj, err := evalAndUnmarshal(fixture, "K8ssandraCluster", &coreapi.K8ssandraCluster{})
+	if obj == nil {
+		return nil, err
 	}
+	return obj.(*coreapi.K8ssandraCluster), err
 }
 
-const fixtureKustomizeTemplate = `apiVersion: kustomize.config.k8s.io/v1beta1
-kind: Kustomization
-namespace: {{.Namespace}}
-resources:
-- ../../../../test/testdata/fixtures/{{ .Fixture }}
-{{if .DcContexts}}
-patches:
-  - patch: |-
-{{range $index, $dcContext := .DcContexts}}
-      - op: replace
-        path: /spec/cassandra/datacenters/{{ $index }}/k8sContext
-        value: {{ $dcContext }}
-{{end}}
-    target:
-      kind: K8ssandraCluster
-{{end}}
-`
+func getFixtureCassandraDatacenter(fixture *TestFixture) (*cassdcapi.CassandraDatacenter, error) {
+	obj, err := evalAndUnmarshal(fixture, "CassandraDatacenter", &cassdcapi.CassandraDatacenter{})
+	if obj == nil {
+		return nil, err
+	}
+	return obj.(*cassdcapi.CassandraDatacenter), err
+}
+
+func getFixtureStargate(fixture *TestFixture) (*stargateapi.Stargate, error) {
+	obj, err := evalAndUnmarshal(fixture, "Stargate", &stargateapi.Stargate{})
+	if obj == nil {
+		return nil, err
+	}
+	return obj.(*stargateapi.Stargate), err
+}
+
+func evalAndUnmarshal(fixture *TestFixture, kind string, obj interface{}) (interface{}, error) {
```

Comment:
Maybe we should make the fixture a first class citizen in the yaml 🤔

Reply:
Hmm what do you mean?